### PR TITLE
Fix Flicker with Dark Theme

### LINKED ISSUE #1711: dark mode pages flicker with white background in iframes
see: https://cal.com/peerrich-eth/orange-dao or  https://cal.com/web3

https://user-images.githubusercontent.com/8019099/152646668-a81c1f40-10a3-4624-a04e-badf006c1f38.mov

### PR DESCRIPTION
**Pull request by hariombalhara**
## What does this PR do?

This is a fix for flicker of Light when using a dark theme.
It also fixes prefers-color-scheme based light and dark themes.

Fixes #1711.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- [x] I checked the refresh behaviour of Availability, EventType pages  with Dark Theme
- [x] I also verified the prefers-color-scheme was honoured on the pages.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code and corrected any misspellings
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by hariombalhara** [comment]
Screenshot 2022-02-08 at 3 47 51 PM
These are some e2e Tests failing which I believe aren't related to my change.

**Comment by arthurdenner** [comment]
@hariombalhara, coincidently, a few minutes before your PR was opened, I opened #1740 which aims to fix the same issue.
Can you check if my changes also fix the issues you attempted to fix?

**Comment by PeerRich** [comment]
> @hariombalhara, coincidently, a few minutes before your PR was opened, I opened #1740 which aims to fix the same issue. Can you check if my changes also fix the issues you attempted to fix?

thank you two for opening a similar Pull Request. do you two wanna find out which solution works best and then close the alternative?

**Comment by arthurdenner** [comment]
@PeerRich, I did fewer changes but I don't know they fix all the issues @hariombalhara attempted to fix with theirs.
I'm leaving for a short vacation today so I can't test but if you find my PR incomplete, feel free to close it. 🤙🏽

File: components/booking/pages/AvailabilityPage.tsx [hunk 1]
```diff
@@ -99,6 +99,7 @@ const AvailabilityPage = ({ profile, eventType, workingHours }: Props) => {
 
   return (
     <>
+      <Theme></Theme>
```
[the remark this answers is not in the document]
Comment (hariombalhara) [reply components/booking/pages/AvailabilityPage.tsx:102]: Yeah. Will make the change 👍

File: lib/hooks/useTheme.tsx [hunk 2]
```diff
@@ -7,19 +8,29 @@ export default function useTheme(theme?: Maybe<string>) {
...
+            __html: `(function (theme) {
+            if (!theme && window.matchMedia("(prefers-color-scheme: dark)").matches) {
+              document.documentElement.classList.add("dark");
+            } else if (!theme) {
+              /** Uncovered case */
+            } else {
+              document.documentElement.classList.add(theme);
+            }
+          })(${themeString})`,
+          }}></script>
```
Comment (zomars) [inline lib/hooks/useTheme.tsx:27]: I can see that this works only on refresh. Can we make this work so we listen for that media query instead and changes display live?

**Comment by zomars** [comment]
Closing in favor of #1740

**Pull request closed without merging**